### ISSUE TITLE
get_geneids.py fails to produce output

### ISSUE THREAD

**Issue by pansapiens** [MonashBioinformaticsPlatform/RNAsik-pipe#30]
```bash
python3 scripts/get_geneids.py --file_type gtf --in_file /references/iGenomes/Mus_musculus/Ensembl/GRCm38/Annotation/Genes/genes.gtf
```
produces no output. This results in downstream Degust-friendly counts files not being generated.

The annotation source doesn't make difference, this effects iGenomes GTFs from UCSF, Enseml and NCBI.

**Comment by pansapiens** [comment]
The issue seems to be that the default `--feat_type` is `gene`, but this feature does not appear in the iGenomes-sourced annotations. `transcript`, `exon` and `CDS` are present.

**Comment by methylnick** [comment]
I from my runnings, it worked for ENSEMBL reference files at least in 1.5.0. UCSC never worked. I suspect one needs to parse some arguements into featurecounts to extract the appropriate annotations. I remember speaking to Kirill about this, and with UCSC there are non-unique identifiers. and it throws an error. 

This is challenging if users require USCS mappings, to align with all their other work and data. Will try to poke around to see if a fix/hack can be done.

**Comment by serine** [comment]
@pansapiens it's been a while, but trying to get up to speed and clear some of these issues. Gonna start with fixing `get_geneids.py` script first. I know you had some suggestion about existing libraries. 

Quickly looked into `scikit-bio` and that doesn't seem to support gtf only gff3, at least this is what the docs say. 

The idea of that script is to get "nice" / extra info for differential expression - Degust file. 

typical output would be 

```
Gene.ID Chrom   Gene.Name       Biotype
gene1   chr1    1500015O10Rik   protein_coding
gene2   chr1    1700015E13Rik   protein_coding
gene3   chr1    1700016C15Rik   protein_coding
gene4   chr1    1700019A02Rik   protein_coding
gene5   chr1    1700019D03Rik   protein_coding
gene6   chr1    1700019O17Rik   protein_coding
gene7   chr1    1700025G04Rik   protein_coding
gene8   chr1    1700056E22Rik   protein_coding
```

As @methylnick mention I need better support for different annotation files

I'm thinking if `scikit-bio` doesn't support gtf should I attempt to add that in?

**Comment by pansapiens** [comment]
PR #38 contains a fixed `get_geneids.py` - although I've discovered this doesn't work on `Homo_sapiens/UCSC/hg38`.

**Comment by serine** [comment]
since I merged your PR #38  @pansapiens I'm closing this issue. 

however I still hope for a better handling of diff gtf and gff files, but for now it's good enough
